### PR TITLE
[13.0] MIG module account_invoice_start_end_dates

### PR DESCRIPTION
**Pull request by alexis-via**

### CODE REVIEW

**Comment by fclementic2c** [comment]
@alexis-via 
I cannot find the menu to launch the cut-of computation.
By the way do not create a specific group for this just use the standard one if you are ok with that.
Thanks

**Comment by alexis-via** [comment]
@fclementic2c This module only contains the start/end date on invoice/move lines ! The menu for cut-off computation is in another module.

File: account_invoice_start_end_dates/i18n/account_invoice_start_end_dates.pot [hunk 1]
```diff
@@ -0,0 +1,131 @@
+# Translation of Odoo Server.
```
Comment (i-vyshnevska) [inline account_invoice_start_end_dates/i18n/account_invoice_start_end_dates.pot:1]: remove this file

File: account_invoice_start_end_dates/models/account_move.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        for move in self:
+            for line in move.line_ids:
+                if line.product_id and line.product_id.must_have_dates:
+                    if not line.start_date or not line.end_date:
+                        raise UserError(_(
+                            "Missing Start Date and End Date for invoice "
+                            "line with Product '%s' which has the "
+                            "property 'Must Have Start and End Dates'.")
+                            % (line.product_id.display_name))
+        return super(AccountMove, self).action_post()
```
Comment (i-vyshnevska) [inline account_invoice_start_end_dates/models/account_move.py]:
```suggestion
        return super().action_post()
```

File: account_invoice_start_end_dates/tests/test_invoice_start_end_dates.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                    'product_id':
+                    self.env.ref('product.product_product_5').id,
+                    'name': 'HD IPBX',
+                    'price_unit': 215.5,
+                    'quantity': 1,
+                    'account_id': self.account_revenue.id,
+                    }),
+                ],
+        })
+        invoice.action_post()
```
Comment (i-vyshnevska) [inline account_invoice_start_end_dates/tests/test_invoice_start_end_dates.py:98]: the test case doesn't cover anything it should create a product with proper flag and test invoice with it

File: account_invoice_start_end_dates/models/account_move.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    def action_post(self):
+        for move in self:
+            for line in move.line_ids:
+                if line.product_id and line.product_id.must_have_dates:
+                    if not line.start_date or not line.end_date:
```
Comment (i-vyshnevska) [inline account_invoice_start_end_dates/models/account_move.py:16]:
```suggestion
                if line..must_have_dates and (not line.start_date or not line.end_date):
```

**Comment by i-vyshnevska** [comment]
@alexis-via you need to run black on PR

**Pull request merged**